### PR TITLE
Changes to fix the question response flow

### PR DESCRIPTION
**Pull request by amanm-alphagov**
Changes have been made to the current flow for question/answer flow for the frontend to reflect the same sequence as seen on https://github.com/alphagov/digital-identity-architecture/blob/cri-kbv-experian-comms/rfc/2022.1-kbv-sequence-diagram.md

### What changed
Refactored the code to move logic into separate methods and added unit tests

### CODE REVIEW

File: lambdas/question/src/main/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandler.java [hunk 1]
```diff
@@ -66,60 +65,15 @@ public QuestionHandler(
...
-                    response.withBody("{\"message\":\"no further questions\"}");
-                }
-            }
+            processQuestionRequest(input);
         } catch (JsonProcessingException jsonProcessingException) {
             LOGGER.error("Failed to parse object using ObjectMapper");
             response.withStatusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
             response.withBody("{ " + ERROR + ":\"" + jsonProcessingException.getMessage() + "\" }");
         } catch (NullPointerException npe) {
+            npe.printStackTrace();
```
Comment (jkunle) [inline lambdas/question/src/main/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandler.java:76]: Please remove this, I forgot to do this

File: lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianService.java [hunk 2]
```diff
@@ -69,8 +69,8 @@ public QuestionAnswerRequest prepareToSubmitAnswers(QuestionState questionState)
                         .collect(Collectors.toList());
 
         questionAnswerRequest.setQuestionAnswers(collect);
-        questionAnswerRequest.setAuthRefNo(questionState.getControl().getAuthRefNo());
-        questionAnswerRequest.setUrn(questionState.getControl().getURN());
+        // questionAnswerRequest.setAuthRefNo(questionState.getControl().getAuthRefNo());
```
Comment (KaraokeStu) [inline lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianService.java:72]: Any particular reason for leaving these in, but commented?

File: lib/src/test/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianServiceTest.java [hunk 3]
```diff
@@ -32,14 +32,13 @@ void setUp() {
...
 
-    @Test
     void shouldConvertQuestionStateToQuestionAnswerRequest() {
         Control controlMock = mock(Control.class);
         when(controlMock.getAuthRefNo()).thenReturn("some-auth-ref");
         when(controlMock.getURN()).thenReturn("some-urn");
 
         QuestionState questionStateMock = mock(QuestionState.class);
-        when(questionStateMock.getControl()).thenReturn(controlMock);
+        // when(questionStateMock.getControl()).thenReturn(controlMock);
```
Comment (KaraokeStu) [inline lib/src/test/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianServiceTest.java:41]: Any particular reason for the commented line still existing?

**Review by KaraokeStu — COMMENTED** [review]
Just a couple of questions around some commented code
Very minor though

File: lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianService.java (hunk 2 above)
Comment (jkunle) [inline lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/ExperianService.java:72]: @amanm-alphagov no long need the `prepareToSubmitAnswers` method

**Review by KaraokeStu — APPROVED** [review]
LGTM

**Pull request merged**